File: Latihan 2/bahasac++/Hardware.cpp
class Hardware : public Product{ //intermediary
	
	//atribut 
	private:
	string brand;
	string model;
	
	//constructor
	public:
	
	Hardware(){
	}
	
	//proses get dan set
	void setBrand(string brand){
		this->brand = brand;
	}
	
	string getBrand(){
		return this->brand;
	}
	
	void setModel(string model){
		this->model = model;
	}
	
	string getModel(){
		return this->model;
	}
	
	//destructor
	~Hardware(){
	}
};
	
	

File: Latihan 2/bahasaphp/Hardware.php
<?php 
 //derived class 1
class Hardware extends Product{
	
	//atribut private
	private $brand;
	private $model;
		
	//constructor
	function __construct(){
	}
		
	//proses get dan set
	function setBrand($brand){
		$this->brand = $brand;
	}
		
	function getBrand(){
		return $this->brand;
	}
		
	function setModel($model){
		$this->model = $model;
	}
		
	function getModel(){
		return $this->model;
	}
	
	//destructor
	 function __destruct(){
	 }
}
?>